skimage/segmentation/tests/test_watershed.py

```diff
@@ -417,10 +424,53 @@ def test_compact_watershed():
                          [1, 1, 1, 2, 2, 2]], dtype=int)
     np.testing.assert_equal(compact, expected)
     normal = watershed(image, seeds)
-    expected = np.ones(image.shape, dtype=int)
-    expected[2, 3:] = 2
+    expected = np.array([[1, 1, 1, 1, 2, 2],
+                         [1, 1, 1, 1, 2, 2],
+                         [1, 1, 1, 1, 2, 2],
+                         [1, 1, 1, 1, 2, 2],
+                         [1, 1, 1, 1, 2, 2]], dtype=int)
     np.testing.assert_equal(normal, expected)
 
+    # checks that compact watershed labels with watershed lines are
+    # a subset of the labels from normal watershed
+    compact_wsl = watershed(image, seeds, compactness=0.01, watershed_line=True)
+    difference = compact_wsl != compact
+    difference[compact_wsl == 0] = False
+
+    assert not np.any(difference)
+
+
+def test_watershed_with_markers_offset():
+    """
+    Regression test from https://github.com/scikit-image/scikit-image/issues/6632
+    Generate an initial image with two overlapping circles.
+    """
```

Comment:
In reply to https://github.com/scikit-image/scikit-image/pull/7071#discussion_r1277815842, thanks for adding the test! Though, I am unsure if I am happy with this I think this actually doesn't cover the case in #6632 qualitatively. To me #6632 is precisely about what happens if one marker spills over into another basin which which I don't think this test covers. So rather than checking if 0 and 4 are identical, I'd be happy with a much simpler case that does qualitatively the same as #6632. E.g. what do you think about checking the output of

```python
image = -np.array(
    [1, 2, 2, 2, 2, 2, 3]
)
markers = np.array(
    [1, 0, 0, 0, 0, 0, 2]
)
ski.segmentation.watershed(image, markers=markers, mask=image != 0)
```
on `main` this gives `[1, 2, 2, 2, 2, 2, 2]`, on your branch this gives `[1, 1, 1, 1, 2, 2, 2]`.

Hope I am not coming across as too nitpicky. I am totally happy to add the tests myself, if you rather focus your time on the actual code. :)

Reply:
@lagru I like your test suggestion but I'd personally like to keep both — the circles are a more complex landscape for the watershed, so maybe it's not too far fetched to imagine an implementation failing one but not both?

Reply:
True :+1:, no reason to waste this test. :D